### PR TITLE
Raiders of the Lost RDP: The Haag Crusade

### PR DESCRIPTION
**Pull request by MHaggis**
# Enhanced RDP Security Monitoring Suite

This PR introduces a comprehensive set of detections and supporting components for monitoring Windows Remote Desktop Protocol (RDP) activities, enhancing our ability to detect potential lateral movement and unauthorized remote access.

### 1. Windows RDP File Execution
- Monitors execution of .rdp files from high-risk directories
- Focuses on temp folders, download directories, and Outlook temporary locations
- Helps identify potential spear-phishing campaigns using malicious RDP files

### 2. Windows RDPClient Connection Sequence Events
- Tracks RDP ClientActiveX connection attempts (Event ID 1024)
- Provides visibility into initial connection sequences
- Helps identify unusual remote access patterns and potential lateral movement

### New Data Source
- Added support for Microsoft-Windows-TerminalServices-RDPClient/Operational logs
- Specifically tracking Event ID 1024 for connection sequence monitoring
- Enhances visibility into RDP client-side activities

### New Macro
- Introduced `wineventlog_rdp` macro for standardized RDP event querying
- Improves consistency across RDP-related detections
- Simplifies future RDP detection development

## Screenshots
- Added detection screenshots for validation and documentation
- Demonstrates expected data presentation and field mapping

### Windows RDP File Execution

image

### Windows RDPClient Connection Sequence Events
image

## Testing
- Validated against Attack Range dataset
- Confirmed detection of RDP connection sequences
- Verified file execution monitoring capabilities

### CODE REVIEW

**Comment by mvelazc0** [comment]
cool use case buddy !

File: detections/endpoint/windows_rdp_file_execution.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+name: Windows RDP File Execution
+id: 0b6b12b9-8ba9-48fe-b3b8-b4e3e1cd22b4
+version: 1
+date: '2024-11-21'
+author: Michael Haag, Splunk
+type: TTP
+status: production
+description: The following analytic detects when a Windows RDP client attempts to execute an RDP file from a temporary directory, downloads directory, or Outlook directories. This detection is significant as it can indicate an attempt for an adversary to delivery a .rdp file, which may be leveraged by attackers to control or exfiltrate data. If confirmed malicious, this activity could lead to unauthorized access, data theft, or further lateral movement within the network. 
+search: '| tstats count min(_time) as firstTime max(_time) as lastTime from datamodel=Endpoint.Processes 
+  where Processes.process IN ("*\\AppData\\Local\\Temp\\*", "*\\AppData\\Local\\Microsoft\\Outlook\\*", "*\\Content.Outlook\\*", "*\\Downloads\\*")
```
Comment (nasbench) [inline detections/endpoint/windows_rdp_file_execution.yml]: Adding the Olk path, as it is used by the new outlook 

```suggestion
  where Processes.process IN ("*\\AppData\\Local\\Temp\\*", "*\\Olk\\Attachments\\*", "*\\AppData\\Local\\Microsoft\\Outlook\\*", "*\\Content.Outlook\\*", "*\\Downloads\\*")
```

File: detections/endpoint/windows_rdp_file_execution.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  by Processes.process Processes.process_name Processes.user Processes.dest Processes.parent_process_name Processes.parent_process
+  | `drop_dm_object_name(Processes)`
+  | `security_content_ctime(firstTime)`
+  | `security_content_ctime(lastTime)`
+  | eval 
+  execution_type=case(
+      match(process, "\\\\Temp\\\\.*\\.(zip|7z|rar|gz|tar|iso|img|vhd|vhdx).*\\.*\\.rdp"), "temp_archive_execution",
+      match(process, "\\\\Downloads\\\\"), "downloads_execution",
+      match(process, "\\\\Temp\\\\"), "temp_execution",
+      match(process, "\\\\Microsoft\\\\Outlook\\\\"), "outlook_execution",
```
Comment (nasbench) [inline detections/endpoint/windows_rdp_file_execution.yml:25]:
```suggestion
      match(process, "\\\\Microsoft\\\\Outlook\\\\"), "outlook_execution",
      match(process, "\\\\Olk\\\\Attachments\\\\"), "outlook_execution",
```

File: detections/endpoint/windows_rdp_file_execution.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+search: '| tstats count min(_time) as firstTime max(_time) as lastTime from datamodel=Endpoint.Processes 
+  where Processes.process IN ("*\\AppData\\Local\\Temp\\*", "*\\AppData\\Local\\Microsoft\\Outlook\\*", "*\\Content.Outlook\\*", "*\\Downloads\\*")
+  AND Processes.process="*.rdp*"
+  by Processes.process Processes.process_name Processes.user Processes.dest Processes.parent_process_name Processes.parent_process
+  | `drop_dm_object_name(Processes)`
+  | `security_content_ctime(firstTime)`
+  | `security_content_ctime(lastTime)`
+  | eval 
+  execution_type=case(
+      match(process, "\\\\Temp\\\\.*\\.(zip|7z|rar|gz|tar|iso|img|vhd|vhdx).*\\.*\\.rdp"), "temp_archive_execution",
```
Comment (nasbench) [inline detections/endpoint/windows_rdp_file_execution.yml]: Adding a couple of other extensions for delivery, namely "tgz" and "cab" 

```suggestion
      match(process, "\\\\Temp\\\\.*\\.(zip|7z|rar|cab|tgz|gz|tar|iso|img|vhd|vhdx).*\\.*\\.rdp"), "temp_archive_execution",
```

**Review by pyth0n1c — APPROVED** [review]
Reviewed and approved after some minor typo updates.
Will merge after CI completes :)

**Pull request merged**